fritz.coffee

```diff
@@ -147,20 +164,76 @@ module.exports = (env) ->
       )
       super()
 
-    # poll device according to interval
-    requestUpdate: ->
+    requestUpdateLegacyMode: ->
+      # Older Fritz!OS versions lower than version 6.20 do not appear to support
+      # the "getDeviceListInfo" service method properly. Thus, the legacyMode is to provide a work
+      # around to supported older Fritz!OS versions
       @plugin.fritzCall("getSwitchState", @config.ain)
-        .then (state) =>
-          @_setState(if state then on else off)
-          @plugin.fritzCall("getSwitchPower", @config.ain)
-          .then (power) =>
-            @_setPower(power)
-            @plugin.fritzCall("getSwitchEnergy", @config.ain)
-            .then (energy) =>
-              @_setEnergy(Math.round(energy / 100.0) / 10.0)
+      .then (state) =>
+        @_setState(if state then on else off)
+        @plugin.fritzCall("getSwitchPower", @config.ain)
+        .then (power) =>
+          @_setPower(power)
+          @plugin.fritzCall("getSwitchEnergy", @config.ain)
+          .then (energy) =>
+            @_setEnergy(Math.round(energy / 100.0) / 10.0)
+      .error (error) ->
+        env.logger.error "Cannot access #{error.options?.url}: #{error.response?.statusCode}"
+
+    # poll device according to interval
+    requestUpdateGetDeviceListInfo:->
+      @plugin.fritzCall("getDeviceListInfo", @config.ain)
+        .then (xmlDeviceInfo) =>
```

Comment:
Indentation on master also varies. See comments above. It is probably due to the IDE. I am using IDEA.